src/test/librbd/mirror/snapshot/test_mock_UnlinkPeerRequest.cc

```diff
@@ -375,7 +377,7 @@ TEST_F(TestMockMirrorSnapshotUnlinkPeerRequest, RemoveSnapshotError) {
 
   C_SaferCond ctx;
   auto req = new MockUnlinkPeerRequest(&mock_image_ctx, snap_id, "peer_uuid",
-                                       &ctx);
+                                       true, &ctx);
```

Comment:
Not seeing a single `MockUnlinkPeerRequest` created with `allow_remove=false` -- need to add coverage for the new code path.

Reply:
I've added the following unit tests: 
`PeerNoRemove`
> goal was to execute code path with single known peer, with allow_remove=false
`UnlinkErrorRestart`
> goal was to execute code path where allow_remove toggles from False to True